Source/WebCore/rendering/CaretRectComputation.cpp

```diff
@@ -91,8 +91,8 @@ static LayoutRect computeCaretRectForEmptyElement(const RenderBoxModelObject& re
         break;
     }
 
-    LayoutUnit x = renderer.borderLeft() + renderer.paddingLeft();
-    LayoutUnit maxX = width - renderer.borderRight() - renderer.paddingRight();
+    LayoutUnit x = renderer.borderAndPaddingStart();
+    LayoutUnit maxX = logicalWidth - renderer.borderAndPaddingEnd();
```

Comment:
have you tested this change with RTL/LTR + text indent? It looks like now x/maxX is logical but some of the code below (alignment) still thinks it is not.

Reply:
Will fix.

Reply:
Fixed and added another test.